### PR TITLE
[AIRFLOW-1695] Add RedshiftHook using boto3

### PR DESCRIPTION
**Pull request by andyxhadji**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1695

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
- Added RedshiftHook using boto3, and tests.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
- Added test_redshift_hook.py - includes unit tests for methods in class

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by bolkedebruin** [comment]
Is this reviewable? If so, please fix your commit message to make it readable. See commit guidelines. Furthermore please make sure the tests actually can run, include the right packages for ci.

**Comment by sid88in** [comment]
thanks @andyxhadji more power to #airflow and #aws

**Comment by andyxhadji** [comment]
@bolkedebruin This is now reviewable, all tests are passing. Please take a look!

**Comment by andyxhadji** [comment]
This is one of the follow-ups to https://github.com/apache/incubator-airflow/pull/2532

File: airflow/contrib/hooks/aws_hook.py [hunk 1]
```diff
@@ -78,35 +78,36 @@ class AwsHook(BaseHook):
     This class is a thin wrapper around the boto3 python library.
     """
 
-    def __init__(self, aws_conn_id='aws_default'):
+    def __init__(self, aws_conn_id='aws_default', region_name=None):
         self.aws_conn_id = aws_conn_id
+        self.region_name = region_name
```
Comment (bolkedebruin) [inline airflow/contrib/hooks/aws_hook.py]: This is obtained from the extra connectin parameters, why have it in the signature?

**Review by bolkedebruin — CHANGES_REQUESTED** [review]
Small nits. I don't think there is a need to touch aws_hook.py

**Comment by bolkedebruin** [comment]
Please also make sure your commit message adheres to the layout (descr lines are too long)

**Comment by andyxhadji** [comment]
@bolkedebruin Fixed - re-running tests because of an unrelated failure (https://travis-ci.org/apache/incubator-airflow/builds/294151528).

**Comment by andyxhadji** [comment]
@bolkedebruin Tests passing!

**Comment by bolkedebruin** [comment]
Your tests are not running as they are not prepended by "test_". Please verify in the travis that your tests run.

**Comment by andyxhadji** [comment]
@bolkedebruin Hmm, I'm not sure if I'm looking in the right place - the tests are prepended with "test_" and the tests run (screenshot from changed files & apache TravisCI below)
image

image

Am I missing something in a file? I really appreciate the help.

**Comment by bolkedebruin** [comment]
Apologies! I did do a search but I probably misspelled.

**Comment by andyxhadji** [comment]
Not a problem! Thanks for the review!

**Pull request closed without merging**